test/fixtures/test-runner/never_ending_async.js

```diff
@@ -0,0 +1,6 @@
+const test = require('node:test');
+const { setTimeout } = require('timers/promises');
+
+test('never ending test', () => {
+  return setTimeout(100_000_000);
+});
```

Comment:
```suggestion

test('never ending test', () => new Promise(() => {});

```

Reply:
this test won't actually be infinite because when no async activity is queued - the test runner ends:
see 
https://github.com/nodejs/node/blob/432d1b50e0432daf7e81dea9a8d6dca64ecde6a4/lib/internal/test_runner/harness.js#L67-L68
and inside `postRun`:
https://github.com/nodejs/node/blob/432d1b50e0432daf7e81dea9a8d6dca64ecde6a4/lib/internal/test_runner/test.js#L348

this design is debatable and might be a little confusing, but out of scope of this PR

this will be the result
```
TAP version 13
# Subtest: never ending test
not ok 1 - never ending test
  ---
  duration_ms: 0.004330489
  failureType: 'cancelledByParent'
  error: 'test did not finish before its parent and was cancelled'
  code: 'ERR_TEST_FAILURE'
  stack: |-
    process.emit (node:events:513:28)
  ...
1..1
# tests 1
# pass 0
# fail 0
# cancelled 1
# skipped 0
# todo 0
# duration_ms 0.236723289
```

Reply:
Leave a comment then?